lib/Backend/arm64/LowerMD.cpp

```diff
@@ -7774,7 +7776,23 @@ LowererMD::FinalLower()
             {
                 IR::BranchInstr *branchInstr = instr->AsBranchInstr();
 
-                if (!LowererMD::IsUnconditionalBranch(branchInstr)) //Ignore BX register based branches & B
+                if (branchInstr->IsMultiBranch())
+                {
+                    Assert(instr->GetSrc1() && instr->GetSrc1()->IsRegOpnd());
+                    IR::MultiBranchInstr * multiBranchInstr = instr->AsBranchInstr()->AsMultiBrInstr();
+
+                    if (multiBranchInstr->m_isSwitchBr &&
+                        (multiBranchInstr->m_kind == IR::MultiBranchInstr::IntJumpTable || multiBranchInstr->m_kind == IR::MultiBranchInstr::SingleCharStrJumpTable))
+                    {
+                        BranchJumpTableWrapper * branchJumpTableWrapper = multiBranchInstr->GetBranchJumpTable();
+                        totalJmpTableSizeInBytes += (branchJumpTableWrapper->tableSize * sizeof(void*));
+
+                        // instrOffset is relative to the end of the function. Jump tables come after the function and so would result in negative offsets. label offsets
+                        // are unsigned so instead give jump table lables offsets relative to the end of the jump table section.
+                        branchJumpTableWrapper->labelInstr->SetOffset(totalJmpTableSizeInBytes);
+                    }
+                }
+                else if (!LowererMD::IsUnconditionalBranch(branchInstr)) //Ignore other direct branches
                 {
```

Comment:
Not sure how jump tables are managed in this case, but typically on ARM/ARM64 we store them as a table of signed PC-relative offsets instead of absolute addresses. Then we fetch the offset from the table, do an ADR of the base, add them together, and jump.